File: android/app/src/test/java/com/phicdy/totoanticipation/model/TeamInfoMapperTest.java
package com.phicdy.totoanticipation.model;

import org.junit.Test;

import static org.hamcrest.CoreMatchers.is;
import static org.hamcrest.MatcherAssert.assertThat;

public class TeamInfoMapperTest {

    @Test
    public void convertEmpty() {
        assertThat(new TeamInfoMapper().fullNameForFootbellGeist(""), is(""));
    }

    @Test
    public void convertInvalidTeam() {
        assertThat(new TeamInfoMapper().fullNameForFootbellGeist("hoge"), is(""));
    }

    @Test
    public void convertKashima() {
        assertThat(new TeamInfoMapper().fullNameForFootbellGeist("鹿島"), is("鹿島アントラーズ"));
    }

    @Test
    public void convertUrawa() {
        assertThat(new TeamInfoMapper().fullNameForFootbellGeist("浦和"), is("浦和レッズ"));
    }

    @Test
    public void convertKofu() {
        assertThat(new TeamInfoMapper().fullNameForFootbellGeist("甲府"), is("ヴァンフォーレ甲府"));
    }

    @Test
    public void convertMarinosu() {
        assertThat(new TeamInfoMapper().fullNameForFootbellGeist("横浜Ｍ"), is("横浜F・マリノス"));
    }

    @Test
    public void convertOmiya() {
        assertThat(new TeamInfoMapper().fullNameForFootbellGeist("大宮"), is("大宮アルディージャ"));
    }

    @Test
    public void convertSendai() {
        assertThat(new TeamInfoMapper().fullNameForFootbellGeist("仙台"), is("ベガルタ仙台"));
    }

    @Test
    public void convertKobe() {
        assertThat(new TeamInfoMapper().fullNameForFootbellGeist("神戸"), is("ヴィッセル神戸"));
    }

    @Test
    public void convertTosu() {
        assertThat(new TeamInfoMapper().fullNameForFootbellGeist("鳥栖"), is("サガン鳥栖"));
    }

    @Test
    public void convertHiroshima() {
        assertThat(new TeamInfoMapper().fullNameForFootbellGeist("広島"), is("サンフレッチェ広島"));
    }

    @Test
    public void convertSeresso() {
        assertThat(new TeamInfoMapper().fullNameForFootbellGeist("Ｃ大阪"), is("セレッソ大阪"));
    }

    @Test
    public void convertGanba() {
        assertThat(new TeamInfoMapper().fullNameForFootbellGeist("Ｇ大阪"), is("ガンバ大阪"));
    }

    @Test
    public void convertKashiwa() {
        assertThat(new TeamInfoMapper().fullNameForFootbellGeist("柏"), is("柏レイソル"));
    }

    @Test
    public void convertIwata() {
        assertThat(new TeamInfoMapper().fullNameForFootbellGeist("磐田"), is("ジュビロ磐田"));
    }

    @Test
    public void convertFCTokyo() {
        assertThat(new TeamInfoMapper().fullNameForFootbellGeist("Ｆ東京"), is("FC東京"));
    }

    @Test
    public void convertKawasaki() {
        assertThat(new TeamInfoMapper().fullNameForFootbellGeist("川崎"), is("川崎フロンターレ"));
    }

    @Test
    public void convertNiigata() {
        assertThat(new TeamInfoMapper().fullNameForFootbellGeist("新潟"), is("アルビレックス新潟"));
    }

    @Test
    public void convertShimizu() {
        assertThat(new TeamInfoMapper().fullNameForFootbellGeist("清水"), is("清水エスパルス"));
    }

    @Test
    public void convertSapporo() {
        assertThat(new TeamInfoMapper().fullNameForFootbellGeist("札幌"), is("コンサドーレ札幌"));
    }

    @Test
    public void convertYamagata() {
        assertThat(new TeamInfoMapper().fullNameForFootbellGeist("山形"), is("モンテディオ山形"));
    }

    @Test
    public void convertMito() {
        assertThat(new TeamInfoMapper().fullNameForFootbellGeist("水戸"), is("水戸ホーリーホック"));
    }

    @Test
    public void convertOita() {
        assertThat(new TeamInfoMapper().fullNameForFootbellGeist("大分"), is("大分トリニータ"));
    }

    @Test
    public void convertMatsumoto() {
        assertThat(new TeamInfoMapper().fullNameForFootbellGeist("松本"), is("松本山雅FC"));
    }

    @Test
    public void convertShonan() {
        assertThat(new TeamInfoMapper().fullNameForFootbellGeist("湘南"), is("湘南ベルマーレ"));
    }

    @Test
    public void convertYamaguchi() {
        assertThat(new TeamInfoMapper().fullNameForFootbellGeist("山口"), is("レノファ山口FC"));
    }

    @Test
    public void convertTokushima() {
        assertThat(new TeamInfoMapper().fullNameForFootbellGeist("徳島"), is("徳島ヴォルティス"));
    }

    @Test
    public void convertFukuoka() {
        assertThat(new TeamInfoMapper().fullNameForFootbellGeist("福岡"), is("アビスパ福岡"));
    }

    @Test
    public void convertSanuki() {
        assertThat(new TeamInfoMapper().fullNameForFootbellGeist("讃岐"), is("カマタマーレ讃岐"));
    }

    @Test
    public void convertChiba() {
        assertThat(new TeamInfoMapper().fullNameForFootbellGeist("千葉"), is("ジェフユナイテッド千葉"));
    }

    @Test
    public void convertVerdy() {
        assertThat(new TeamInfoMapper().fullNameForFootbellGeist("東京Ｖ"), is("東京ヴェルディ1969"));
    }

    @Test
    public void convertKanazawa() {
        assertThat(new TeamInfoMapper().fullNameForFootbellGeist("金沢"), is("ツエーゲン金沢"));
    }

    @Test
    public void convertMachida() {
        assertThat(new TeamInfoMapper().fullNameForFootbellGeist("町田"), is("町田ゼルビア"));
    }

    @Test
    public void convertKumamoto() {
        assertThat(new TeamInfoMapper().fullNameForFootbellGeist("熊本"), is("ロアッソ熊本"));
    }

    @Test
    public void convertGunma() {
        assertThat(new TeamInfoMapper().fullNameForFootbellGeist("群馬"), is("ザスパクサツ群馬"));
    }

    @Test
    public void convertYokomaFC() {
        assertThat(new TeamInfoMapper().fullNameForFootbellGeist("横浜Ｃ"), is("横浜FC"));
    }

    @Test
    public void convertNagoya() {
        assertThat(new TeamInfoMapper().fullNameForFootbellGeist("名古屋"), is("名古屋グランパス"));
    }

    @Test
    public void convertGifu() {
        assertThat(new TeamInfoMapper().fullNameForFootbellGeist("岐阜"), is("FC岐阜"));
    }

    @Test
    public void convertKyoto() {
        assertThat(new TeamInfoMapper().fullNameForFootbellGeist("京都"), is("京都サンガF.C."));
    }

    @Test
    public void convertOkayama() {
        assertThat(new TeamInfoMapper().fullNameForFootbellGeist("岡山"), is("ファジアーノ岡山"));
    }

    @Test
    public void convertEhime() {
        assertThat(new TeamInfoMapper().fullNameForFootbellGeist("愛媛"), is("愛媛FC"));
    }

    @Test
    public void convertNagasaki() {
        assertThat(new TeamInfoMapper().fullNameForFootbellGeist("長崎"), is("V・ファーレン長崎"));
    }
}

File: android/app/src/test/java/com/phicdy/totoanticipation/model/TeamInfoMapperTest.kt
package com.phicdy.totoanticipation.model

import org.hamcrest.CoreMatchers.`is`
import org.hamcrest.MatcherAssert.assertThat
import org.junit.experimental.theories.DataPoints
import org.junit.experimental.theories.Theories
import org.junit.experimental.theories.Theory
import org.junit.runner.RunWith

@RunWith(Theories::class)
class TeamInfoMapperTest {

    companion object {
        @DataPoints
        @JvmField
        var gameAndResult = listOf(
                Pair("", ""),
                Pair("hoge", ""),
                Pair("鹿島", "鹿島アントラーズ"),
                Pair("浦和", "浦和レッズ"),
                Pair("甲府", "ヴァンフォーレ甲府"),
                Pair("横浜Ｍ", "横浜F・マリノス"),
                Pair("大宮", "大宮アルディージャ"),
                Pair("仙台", "ベガルタ仙台"),
                Pair("神戸", "ヴィッセル神戸"),
                Pair("鳥栖", "サガン鳥栖"),
                Pair("広島", "サンフレッチェ広島"),
                Pair("Ｃ大阪", "セレッソ大阪"),
                Pair("Ｇ大阪", "ガンバ大阪"),
                Pair("柏", "柏レイソル"),
                Pair("磐田", "ジュビロ磐田"),
                Pair("Ｆ東京", "FC東京"),
                Pair("川崎", "川崎フロンターレ"),
                Pair("新潟", "アルビレックス新潟"),
                Pair("清水", "清水エスパルス"),
                Pair("札幌", "コンサドーレ札幌"),
                Pair("山形", "モンテディオ山形"),
                Pair("水戸", "水戸ホーリーホック"),
                Pair("大分", "大分トリニータ"),
                Pair("松本", "松本山雅FC"),
                Pair("湘南", "湘南ベルマーレ"),
                Pair("山口", "レノファ山口FC"),
                Pair("徳島", "徳島ヴォルティス"),
                Pair("福岡", "アビスパ福岡"),
                Pair("讃岐", "カマタマーレ讃岐"),
                Pair("千葉", "ジェフユナイテッド千葉"),
                Pair("東京Ｖ", "東京ヴェルディ1969"),
                Pair("金沢", "ツエーゲン金沢"),
                Pair("町田", "町田ゼルビア"),
                Pair("熊本", "ロアッソ熊本"),
                Pair("群馬", "ザスパクサツ群馬"),
                Pair("横浜Ｃ", "横浜FC"),
                Pair("名古屋", "名古屋グランパス"),
                Pair("岐阜", "FC岐阜"),
                Pair("京都", "京都サンガF.C."),
                Pair("岡山", "ファジアーノ岡山"),
                Pair("愛媛", "愛媛FC"),
                Pair("長崎", "V・ファーレン長崎"),
                Pair("岩手", "グルージャ盛岡"),
                Pair("八戸", "ヴァンラーレ八戸"),
                Pair("琉球", "FC琉球"),
                Pair("栃木", "栃木SC"),
                Pair("鹿児島", "鹿児島ユナイテッドFC")
        )
    }

    @Theory
    fun `when map team then return the team in FootballGeist`(mapping: Pair<String, String>) {
        assertThat(TeamInfoMapper().fullNameForFootbellGeist(mapping.first), `is`(mapping.second))
    }

}
